src/main/java/agh/ics/oop/AbstractWorldMap.java

```diff
@@ -2,34 +2,35 @@
 
 import java.util.*;
 
-public abstract class AbstractWorldMap implements IWorldMap, IPositionChangeObserver {
+public abstract class AbstractWorldMap implements IWorldMap {
 
     protected final HashMap<Vector2d, IMapElement> elements;
 
     public AbstractWorldMap() {
         this.elements = new HashMap<>();
     }
 
+    public Map<Vector2d, IMapElement> getElements() {
+        return Collections.unmodifiableMap(this.elements);
+    }
+
     public boolean canMoveTo(Vector2d position) {
         return !(this.objectAt(position) instanceof Animal);
     }
 
     public boolean place(Animal animal) {
```

Comment:
skoro rzucamy wyjątkiem to nie zwrócimy tu nigdy false -> ta metoda nie powinna już zwracać boolean